client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
I do not think you understand what I was talking about. There is a done chan without buffer. If you break out at line 353, then the other go routine will block at line 347.

Also I still do not quite understand your explanation. When does body.Close return an error?